src/providers/krb5/krb5_child.c

```diff
@@ -1221,34 +1264,18 @@ static krb5_error_code sss_krb5_responder(krb5_context ctx,
         for (c = 0; question_list[c] != NULL; c++) {
             DEBUG(SSSDBG_TRACE_ALL, "Got question [%s].\n", question_list[c]);
 
+            /* It is expected that the answer_*() functions only return EOK
+             * (success) if the authentication was successful, i.e. during
+             * SSS_PAM_AUTHENTICATE. In all other cases, e.g. during
+             * SSS_PAM_PREAUTH either EAGAIN should be returned to indicate
+             * that the other available authentication methods should be
+             * checked as well. Or some other error code to indicate a fatal
+             * error where no other methods should be tried.
```

Comment:
What does it mean if an answer_* function returns EOK during SSS_PAM_PREAUTH? Or is that no longer possible with these changes?

Reply:
Hi,

my understanding was that you've added `EAGAIN` as a return code to indicate that other methods should be checked as well. Since we want to collect all available methods during `SSS_PAM_PREAUTH` and the `answer_*` cannot know if it is the last one to be called they all have to return `EAGAIN` during `SSS_PAM_PREAUTH`.

As an alternative, it would be possible to handle `EOK` and `EAGAIN` differently in `sss_krb5_responder()` so that during `SSS_PAM_PREAUTH` a return value of `EOK` will trigger continue and all other codes will be an error and for `SSS_PAM_AUTHENTICATE` `EAGAIN` will trigger continue and all other codes will end the processing.

Which one would you prefer?

bye,
Sumit

Reply:
I think it's okay as-is. Thanks for the explanation.